_im-plugin/performance.md

```diff
@@ -0,0 +1,134 @@
+---
+layout: default
+title: Performance considerations
+nav_order: 41
+has_children: false
+---
+
+# Performance considerations while tuning Opensearch for indexing speed
+
+The following configurations have demonstrated around 60% improvements in throughput when
+running an indexing only workload compared to the out of box experience. The workload did not
+incorporate search or other scenarios. Only the opensearch server process was run on the machines,
+with the benchmark clients hosted on a different node.
+
+The execution environment used comprises of Intel EC2 instances (r7iz.2xlarge) on the AWS cloud, and
+workload was StackOverflow dataset available as part of OpenSearch Benchmark.
+
+## Java Heap Size
+
+A larger Java heap size is useful for indexing. Setting the Java min and max heap sizes to 50% of RAM
+sizes shows better indexing performance on ec2 instances.
+
+## Flush translog threshold
+
+Default value for `flush_threshold_size` is 512MB. This means that the translog is flushed when it reaches 512 MB. The weight of the indexing load determines the frequency of the translog. When you increase index.translog.flush_threshold_size, the node performs the translog operation less frequently. Because OpenSearch Service flushes are resource-intensive operations, reducing the frequency of translogs improves indexing performance. By increasing the flush threshold size, the OpenSearch Service cluster also creates fewer large segments (instead of multiple small segments). Large segments merge less often, and more threads are used for indexing instead of merging.
+
+For pure indexing workloads, consider increasing this value, e.g., 25% of the Java Heap size, to improve indexing performance.
+
+Note: An increase in index.translog.flush_threshold_size can also increase the time that it takes for a translog to complete. If a shard fails, then recovery takes more time because the translog is larger.
```

Comment:
```suggestion
An increased `index.translog.flush_threshold_size` can also increase the time that it takes for a translog to complete. If a shard fails, then recovery takes more time because the translog is larger.
{: .note}
```